src/lib.rs

```diff
@@ -226,7 +226,7 @@ pub enum SchemeData {
 }
 
 /// Components for URLs in a *relative* scheme such as HTTP.
-#[derive(PartialEq, Eq, Clone, Debug, Hash, PartialOrd, Ord)]
+#[derive(Clone, Debug, Hash, PartialOrd, Ord)]
```

Comment:
BTW I've filed the `derive(Hash)` issue against rustc: https://github.com/rust-lang/rust/issues/29758